test/hotspot/jtreg/compiler/loopopts/superword/TestDependencyOffsets.java

```diff
@@ -486,15418 +409,475 @@
+
+        /*
+         * Template for init method generation.
+         */
+        String generateInit() {
+            return String.format("""
+                       static void init(%s[] a, %s[] b) {
+                           for (int i = 0; i < SIZE; i++) {
+                               a[i] = (%s)(2 * i);
+                               b[i] = (%s)(3 * i);
+                           }
+                       }
+                   """,
+                   name, name, name, name);
+        }
+
+        /*
+         * Template for verify method generation.
+         */
+        String generateVerify() {
+            return String.format("""
+                       static void verify(String context, %s[] aTest, %s[] bTest, %s[] aGold, %s[] bGold) {
+                           for (int i = 0; i < SIZE; i++) {
+                               if (aTest[i] != aGold[i] || bTest[i] != bGold[i]) {
+                                   throw new RuntimeException("Wrong result in " + context + " at i=" + i + ": " +
+                                                              "aTest=" + aTest[i] + ", aGold=" + aGold[i] +
+                                                              "bTest=" + bTest[i] + ", bGold=" + bGold[i]);
+                               }
+                           }
+                       }
+                   """,
+                   name, name, name, name);
+        }
+    }
+
+    static Type[] types = new Type[] {
+        new Type("int",    4, "-11",    "*", "MUL_VI"),
+        new Type("long",   8, "-11",    "+", "ADD_VL"), // aarch64 NEON does not support MulVL
+        new Type("short",  2, "-11",    "*", "MUL_VS"),
+        new Type("char",   2, "-11",    "*", "MUL_VS"), // char behaves like short
+        new Type("byte",   1, "11",     "*", "MUL_VB"),
+        new Type("float",  4, "1.001f", "*", "MUL_VF"),
+        new Type("double", 8, "1.001",  "*", "MUL_VD"),
+    };
+
+    static List<Integer> getOffsets() {
+        // Some carefully hand-picked values
+        int[] always = new int[] {
+            0,
+            -1, 1,
+            -2, 2,     // 2^1
+            -3, 3,
+            -4, 4,     // 2^2
+            -7, 7,
+            -8, 8,     // 2^3
+            -14, 14,
+            -16, 16,   // 2^4
+            -18, 18,
+            -20, 20,
+            -31, 31,
+            -32, 32,   // 2^5
+            -63, 63,
+            -64, 64,   // 2^6
+            -65, 65,
+            -128, 128, // 2^7
+            -129, 129,
+            -192, 192, // 3 * 64
+        };
+        Set<Integer> set = Arrays.stream(always).boxed().collect(Collectors.toSet());
+
+        // Sample some random values on an exponental scale
+        for (int i = 0; i < 10; i++) {
+            int base = 4 << i;
+            int offset = base + RANDOM.nextInt(base);
+            set.add(offset);
+            set.add(-offset);
+        }
+
+        List<Integer> offsets = new ArrayList<Integer>(set);
+        return offsets;
```

Comment:
Can be simplified to:
```suggestion
        return new ArrayList<>(set);
```